README.md

```diff
@@ -198,6 +198,160 @@ Similarly to derived stores, addtional load functionality is bundled with readab
+
+If an external source updates the storage item of the persisted store the two values will go out of sync. In such a case we can call `.resync()` on the store in order to update the store the *parsed* value of the storage item.
+
+#### persisted configuration / consent
+
+Persisting data to storage or cookies is subject to privacy laws regarding consent in some jurisdictions. Instead of building two different data flows that depend on whether tracking consent has been given or not, you can instead configure your persisted stores to work in both cases. To do so you will need to call the `configurePersistedConsent` function and pass in a consent checker that will accept a `consent level` and return a boolean indicating whether your user has consented to that level of tracking. You can then provide a consent level when building your persisted stores that will be passed to to the checker before storing data.
+
+*Supporting tracking consent is simple...*
+
+```javascript
+configurePersistedConsent(
+  (consentLevel) =>  window.consentLevels.includes(consentLevel);
+);
+
+const hasDismissedTooltip = persisted(
+  false, 
+  'TOOLTIP_DISMISSED', 
+  { 
+    storageType: StorageType.COOKIE,
+    consentLevel: 'TRACKING'
+  }
+);
+```
+
+Here we hypothesize a setup where a user's consentLevels are accessible through the window object. We would like to track the dismissal of a tooltip and ideally persist that across page loads. To do so we set up a `hasDismissedTooltip` store that can bet set like any other writable store. If the user has consented to the `TRACKING` consent level, then setting the store will also set a `TOOLTIP_DISMISSED` cookie. Otherwise no data will be persisted and the store will initialize to the default value `false` on each page load.
+
+Note that if no consent level is provided, `undefined` will be passed to the consent checker. This can be handled to provide a default consent for your persisted stores when a consent level is not provided.
+
+### state (BETA)
+
+State stores are a kind of non-Loadable Readable store that can be generated alongside async stores in order to track their load state. This can be done by passing the `trackState` to the store options during creation. This is particular useful for reloadable or asyncDerived stores which might go into a state of pulling new data.
+
+*State stores can be used to conditionally render our data...*
+
+```javascript
+<script>
+  let searchInput;
+  const searchTerms = writable();
+  const {store: searchResults, state: searchState} = asyncDerived(
+    searchTerms,
+    async ($searchTerms) => {
+      const response = await search($searchTerms);
+      return response.results;
+    },
+    { trackState: true }
+  )
+</script>
+  <input bind:value={searchInput}>
+  <button on:click={() => searchTerms.set(searchInput)}>search</button>
+  {#if $searchState === LoadState.LOADING}
```

Comment:
personally more a fan of string unions for these, so you dont have to import the `enum` every time you wanna use it. (and still get good type safety)

Reply:
Good idea

Reply:
@Akolyte01  just need to make sure all stores are typed correctly to `LoadState` not just raw `string`

<img width="668" alt="image" src="https://user-images.githubusercontent.com/9781750/196268286-2066e2f8-e0ae-4ae3-a527-0298557767b7.png">

Reply:
Completion looks good too 
<img width="568" alt="image" src="https://user-images.githubusercontent.com/9781750/196268423-0cd8ed61-8c9f-4e99-a59d-49d94932c7ff.png">

Reply:
took a stab here lemme know if its same as what you saw before - https://github.com/square/svelte-store/pull/34

Reply:
I think the root of the problem is that svelte templates do not have a good concept of types, even if using typescript in the script section.  I suppose using an enum does not fix that issue.

Reply:
@smblee Could you rebase that PR? Then I am good with merging

Reply:
Ah i like this 👍 a lot more explicit